src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1277 @@
+            var peerB = new Peer
+            {
+                AgentVersion = "peerB",
+                Id = bid,
+                PublicKey =
+                    "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE=",
+                Addresses = new MultiAddress[]
+                {
+                    $"/ip4/127.0.0.2/tcp/2/ipfs/{bid}",
+                    $"/ip4/127.0.0.3/tcp/3/ipfs/{bid}"
+                }
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB};
+            await swarmB.StartAsync();
+            var _ = await swarmB.StartListeningAsync("/ip4/127.0.0.1/tcp/0");
+            Assert.IsTrue(peerB.Addresses.Any());
+
+            var swarm = new SwarmService {LocalPeer = _self};
+            await swarm.StartAsync();
+            try
+            {
+                var remotePeer = (await swarm.ConnectAsync(peerB)).RemotePeer;
+                Assert.IsNotNull(remotePeer.ConnectedAddress);
+                Assert.AreEqual(peerB.PublicKey, remotePeer.PublicKey);
+                Assert.IsTrue(remotePeer.IsValid());
+                Assert.IsTrue(swarm.KnownPeers.Contains(peerB));
+            }
+            finally
+            {
+                await swarm.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task ConnectionEstablished()
+        {
+            var peerB = new Peer
+            {
+                AgentVersion = "peerB",
+                Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+                PublicKey =
+                    "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB};
+            var swarmBConnections = 0;
+            swarmB.ConnectionEstablished += (s, e) => { ++swarmBConnections; };
+            await swarmB.StartAsync();
+            var peerBAddress = await swarmB.StartListeningAsync("/ip4/127.0.0.1/tcp/0");
+
+            var swarm = new SwarmService {LocalPeer = _self};
+            var swarmConnections = 0;
+            swarm.ConnectionEstablished += (s, e) => { ++swarmConnections; };
+            await swarm.StartAsync();
+            try
+            {
+                var _ = await swarm.ConnectAsync(peerBAddress);
+                Assert.AreEqual(1, swarmConnections);
+
+                // wait for swarmB to settle
+                var endTime = DateTime.Now.AddSeconds(3);
+                while (true)
+                {
+                    if (DateTime.Now > endTime)
+                    {
+                        Assert.Fail("swarmB did not raise event.");
+                    }
+
+                    if (swarmBConnections == 1)
+                    {
+                        break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979568)